pkg/devfile/detect.go

```diff
@@ -140,33 +140,37 @@ func AnalyzePath(a Alizer, localpath, context, devfileRegistryURL string, devfil
 		if dockerfileUri, err := SearchForDockerfile(devfile); err != nil {
 			return err
 		} else if len(dockerfileUri) > 0 {
+			dockerfileContextMapFromRepo[context] = path.Join(context, dockerfileUri)
 			isDockerfilePresent = true
 		}
 	}
 
 	if !isDockerfilePresent {
 		// if we didnt find any devfile/dockerfile upto our desired depth, then use alizer
-		devfile, detectedDevfileEndpoint, detectedSampleName, err := AnalyzeAndDetectDevfile(a, localpath, devfileRegistryURL)
+		detectedDevfile, detectedDevfileEndpoint, detectedSampleName, err := AnalyzeAndDetectDevfile(a, localpath, devfileRegistryURL)
 		if err != nil {
 			if _, ok := err.(*NoDevfileFound); !ok {
 				return err
 			}
 		}
 
-		if !isDevfilePresent && len(devfile) > 0 {
-			// If a devfile is not present at this stage, just update devfileMapFromRepo and devfilesURLMapFromRepo
-			// Dockerfile is not needed because all the devfile registry samples will have a Dockerfile entry
-			devfileMapFromRepo[context] = devfile
-			devfilesURLMapFromRepo[context] = detectedDevfileEndpoint
-		} else if isDevfilePresent && len(devfile) > 0 {
-			// If a devfile is present but we could not determine a dockerfile, then update dockerfileContextMapFromRepo
+		if len(detectedDevfile) > 0 {
```

Comment:
Nit: Couldn't we still keep it as `if !isDevfilePresent && len(detectedDevfile) > 0`?

Reply:
there is also some code after `if !isDevfilePresent {}` block which is executed for when both a devfile is present and not present:

1. Devfile NOT Present: Set the Alizer detected Devfile
2. Devfile NOT Present & Devfile Present: Set the Dockerfile full URL

Reply:
```
if len(detectedDevfile) > 0 {
			if !isDevfilePresent {
				// Save the Alizer detected Devfile
			}
			
                         // Save the Dockerfile full URL
			isDockerfilePresent = true
		}
```

Reply:
I clearly cannot read lol. Okay that makes sense, dunno why I missed the code underneath.